projects/project3/src/main/scala/blast/main/Main.scala

```diff
@@ -0,0 +1,94 @@
+package blast.main
+
+import org.apache.spark.sql.SparkSession
+import org.apache.jena.riot.Lang
+import org.apache.spark.rdd.RDD
+import org.apache.hadoop.hdfs.DFSClient
+import DataStructures.{Attribute, DatasetReader, EntityProfile, IdDuplicates}
+import blast.AttributeSchema.AttributeProfile
+import blast.AttributeSchema.AttributeMatchInduction
+import blast.Blocking.Blocker
+import blast.Blocking.MetaBlocker
+import blast.data_processing.read_GroundTruth
+
+import scala.collection.JavaConverters._
+
+
+
+
+object Main {
+
+  def convertFile(spark : SparkSession, inputP:String, outputP: String): Unit = {
+    val ds:RDD[EntityProfile] = spark.sparkContext.parallelize(DatasetReader.readDataset(inputP))
```

Comment:
My previous comment is related to this. Instead of reading them sequentially and then parallelize them using Spark, you could read them using SparkContext instead. I think we discussed this during our last meeting.